WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController+Extensions.swift

```diff
@@ -662,8 +661,229 @@ private extension NotificationDetailsViewController
 
 
 
-/// UITextViewDelegate
-///
+// MARK: - Media Download Helpers
+//
+extension NotificationDetailsViewController
+{
+    func downloadAndResizeMedia(indexPath: NSIndexPath, blockGroup: NotificationBlockGroup) {
+        //  Notes:
+        //  -   We'll *only* download Media for Text and Comment Blocks
+        //  -   Plus, we'll also resize the downloaded media cache *if needed*. This is meant to adjust images to
+        //      better fit onscreen, whenever the device orientation changes (and in turn, the maxMediaEmbedWidth changes too).
+        //
+        let imageUrls = blockGroup.imageUrlsForBlocksOfTypes(Media.richBlockTypes)
+
+        let completion = {
+            // Workaround: Performing the reload call, multiple times, without the .BeginFromCurrentState might
+            // lead to a state in which the cell remains not visible.
+            //
+            UIView.animateWithDuration(Media.duration, delay: Media.delay, options: Media.options, animations: { [weak self] in
+                self?.tableView.reloadRowsAtIndexPaths([indexPath], withRowAnimation: .Fade)
+            }, completion: nil)
+        }
+
+        mediaDownloader.downloadMedia(urls: imageUrls, maximumWidth: maxMediaEmbedWidth, completion: completion)
+        mediaDownloader.resizeMediaWithIncorrectSize(maxMediaEmbedWidth, completion: completion)
+    }
+
+    var maxMediaEmbedWidth: CGFloat {
+        let textPadding = NoteBlockTextTableViewCell.defaultLabelPadding
+        let portraitWidth = WPDeviceIdentification.isiPad() ? WPTableViewFixedWidth : view.bounds.width
```

Comment:
Definitely not. But we'll be addressing Adaptivity in other PR (that task is already in #5613 punch list), otherwise, the footprint of this one is gonna go up like crazy.

Thanks Will!